docs/dotnet-five.md

```diff
@@ -0,0 +1,148 @@
+
+- Records: Immutable reference types that behave like value types, and introduce the new `with` keyword into the language.
+- Relational pattern matching: Extends pattern matching capabilities to relational operators for comparative evaluations and expressions, including logical patterns - new keywords `and`, `or`, and `not`.
+- Top-level statements: As a means for accelerating adoption and learning of C#, the `Main` method can be omitted and application as simple as the following is valid:
+
+   ```csharp
+   System.Console.Write("Hello world!");
+   ```
+
+- Functional pointers: Language constructs that expose intermediate language (IL) the following opcodes `ldftn` and `calli`.
+
+For more information on the available C# 9 features, see [What's new in C# 9](csharp/whats-new/csharp-9.md).
+
+#### Source generators
+
+In addition to some of the highlighted new C# features, source generators are making their way into developer projects. Source generators allow code that runs during compilation to inspect your program and produce additional files that are compiled together with the rest of your code.
+
+For more information on source generators, see [Introducing C# source generators](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators) and [C# source generator samples](https://devblogs.microsoft.com/dotnet/new-c-source-generator-samples).
+
+### F# updates
+
+F# is the .NET functional programming language, and with .NET 5, developers have access to F# 5. Here are several new features of F# 5, for more information, see [What's new in F# 5](fsharp/whats-new/fsharp-50.md).
+
+#### Interpolated strings
+
+Similar to interpolated string in C#, and even JavaScript - F# supports basic string interpolation.
+
+```fsharp
+let name = "David"
+let age = 36
+let message = $"{name} is {age} years old."
+```
+
+In addition to basic string interpolation, there is typed interpolation. With typed interpolation, a given type must match the format specifier.
+
+```fsharp
+let name = "David"
+let age = 36
+let message = $"%s{name} is %d{age} years old."
+```
+
+This is similar to the [`sprintf`](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-core-printfmodule.html#sprintf) function that formats a string based on type-safe inputs.
+
+### Visual Basic updates
+
+There are no new language features for Visual Basic in .NET 5. However, with .NET 5, Visual Basic support is extended to:
+
+- Console Application `console`
+- Class library `classlib`
+- WPF Application `wpf`
+- WPF Class library `wpflib`
+- WPF Custom Control Library `wpfcustomcontrollib`
+- WPF User Control Library `wpfusercontrollib`
+- Windows Forms (WinForms) Application `winforms`
+- Windows Forms (WinForms) Class library `winformslib`
+- Unit Test Project `mstest`
+- NUnit 3 Test Project `nunit`
+- NUnit 3 Test Item `nunit-test`
+- xUnit Test Project `xunit`
```

Comment:
It's not obvious what the code-style text is following the description -- this might be clearer as a table with header text for column 1 (Description) and 2 (dotnet new parameter).